workflow/setup_xml.py

```diff
@@ -63,6 +90,12 @@ def main(*argv):
 
     check_expdir(user_inputs.expdir, base['EXPDIR'])
 
+    # Check if "HOMEDIR","STMP","PTMP" dirrctories are writable
+    dirKeys = ["HOMEDIR","STMP","PTMP"]
+    for dk in dirKeys:
+        print(f'{dk}:')
+        check_dir_writable(base[dk])
```

Comment:
Check the return value of `check_dir_writable` and raise an informatve error:

```suggestion
        if not check_dir_writable(base[dk]):
            raise PermissionError(f`The {dk} path {base[dk]} cannot be written to!  Please correct this path and try again`)
```

Reply:
@DavidHuber-NOAA 
Thank you very much for your comments and suggestions. I agree with your suggestions and have modified the codes.